listener/src/main/java/org/apache/geode/examples/listener/Example.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.examples.listener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.Random;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+import org.apache.geode.cache.CacheListener;
+import org.apache.geode.cache.EntryEvent;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+
+public class Example implements Consumer<Region<Integer, String>> {
+  public static final int ITERATIONS = 100;
+
+  final List<CacheListener> cacheListeners = new ArrayList();
```

Comment:
Does CacheListeners really need to be a list in this example?

Reply:
Happy to make it private. Is this a Geode style? I recall that the Java style was package-protected by default.

I made it a list because the Javadocs for CacheListeners says they're called in the order in which they're added, which implies ordering matters.

Reply:
The java language defaults to making your fields package scope ... but it's rare that you actually *want* to do that. Generally fields should be private and final if possible.

For list - maybe I'm misreading. Your example only has 1 cache listener, right?

Reply:
I've made it private to be consistent with Geode. (But I learned to use package private default from the writings of Gosling & Joy. #appealtoauthority)

Good point. At one point there were more listeners but since there is only the one I've removed the list from Example.